internal/db/migrations.go

```diff
@@ -18,16 +20,37 @@ const migrationVersion = 1
 //go:embed migrations/*.sql
 var migrations embed.FS
 
-//runMigrations migrates the postgres schema to the current version
-func runMigrations(db *sql.DB, logger *zap.Logger) {
+//getNewMigration get the new migrate instance 
+func getNewMigration(db *sql.DB) (source.Driver, database.Driver, *migrate.Migrate, error) {
```

Comment:
Rename the function to `buildMigrationClient`. Also, why are we returning the source here?

Reply:
1. I will change the name to buildMigrationClient
2. we need Source and target to close the connection in the up/downmigration()  otherwise not able to run the migration in closed connection